src/main/java/eventprocessing/responseservices/ResponseService.java

```diff
@@ -0,0 +1,29 @@
+package eventprocessing.responseservices;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import eventprocessing.models.Message;
+import eventprocessing.models.Response;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class ResponseService {
+
+    private final Logger logger = LogManager.getLogger("ResponseService");
+
+    public Response parseResponse(String jsonString) {
+        Gson gson = new Gson();
+        Response responseObject = null;
+        try {
+            responseObject = gson.fromJson(jsonString, Response.class);
+            String messageString = responseObject.getMessageString();
+            Message messageObject = gson.fromJson(messageString, Message.class);
+            responseObject.setMessage(messageObject);
```

Comment:
This is quite hard to follow - is it possible to deserialize everything at once, so you don't have to call this setter directly?

Reply:
Unfortunately not because the form we received it is not valid JSON. The field we want to extract - "message" - is not a valid JSON but is structured as one.